src/main/java/net/mcreator/plugin/Plugin.java

```diff
@@ -131,6 +136,18 @@ public String getPluginVersion() {
 		return info.getVersion();
 	}
 
+	public URL toURL() throws MalformedURLException {
+		if (file.isDirectory()) {
+			return (file.toURI().toURL());
+		} else {
+			return (new URL("jar:file:" + file.getAbsolutePath() + "!/"));
+		}
+	}
+
+	@Nullable public String getJavaPlugin() {
+		return javaplugin;
+	}
+
 	@Override public String toString() {
 		return "Plugin{" + "file=" + file + ", id='" + id + '\'' + '}';

```

Comment:
This looks weird.
```suggestion
		return "Plugin{file=" + file + ", id='" + id + "'}";
```

Reply:
This was as this before, maybe another PR